addons/website_event/static/src/client_action/website_event_ticket_registration_dialog.js

```diff
@@ -0,0 +1,100 @@
+/** @odoo-module **/
+
+import { Component, useState } from "@odoo/owl";
+import { Dialog } from "@web/core/dialog/dialog";
+import { useService } from "@web/core/utils/hooks";
+
+export class WebsiteEventTicketRegistrationDialog extends Component {
+    static template = "website_event.WebsiteEventTicketRegistration";
+    static components = { Dialog };
+    static props = {
+        close: Function,
+        data: Object,
+    };
+
+    setup() {
+        this.orm = useService("orm");
+        this.registrationCount = 0;
+        this.registrations = useState(this.buildRegistrations());
+        this.registrationsToDelete = useState([]);
+    };
+
+    get data() {
+        return this.props.data;
+    };
+
+    buildRegistrations(){
+        let registrations = [];
+        if (this.props.data.registrations) {
+            this.props.data.registrations.forEach(registration => {
+                this.registrationCount++;
+                registrations.push({
+                    index: this.registrationCount,
+                    registrationId: registration.registration_id,
+                    ticketTypeId: registration.ticket_id,
+                    ticketTypeName: registration.ticket_name,
+                    answers: registration.answers,
+                });
+            });
+        }
+        else {
+            this.props.data.tickets.forEach(ticketType => {
+                for (let i = 1; i <= ticketType.quantity; i++) {
+                    this.registrationCount++;
+                    registrations.push({
+                        index: this.registrationCount,
+                        ticketTypeId: ticketType.id,
+                        ticketTypeName: ticketType.name,
+                        answers: Object.assign({}, ...this.props.data.event.specific_question_ids.map((question) => (
+                            { [question.id]: (this.registrationCount === 1 && this.props.data.default_first_attendee !== undefined) ? this.props.data.default_first_attendee[question.question_type] : '' }
+                        ))),
+                    })
+                }
+            });
+        }
+        return registrations;
+    }
+
+    getAvailableTicketTypes(actualTicketTypeId) {
+        return this.props.data.event.tickets_ids.filter((ticket) => ticket.seats_available > 0 && ticket.id !== actualTicketTypeId);
+    };
+
+    addRegistration() {
+        this.registrationCount++;
+        this.registrations.push({
+            index: this.registrationCount,
+            ticketTypeId: this.props.data.event.tickets_ids[0].id,
```

Comment:
If magic ticket -> Select ticket type is hidden with 'd-none' 
when adding new ticket, we use the data from the first registration (If only one registration remove the delete button to avoid empty list)